src/track/track.cpp

```diff
@@ -327,7 +327,25 @@ void Track::setReplayGain(const mixxx::ReplayGain& replayGain) {
     QMutexLocker lock(&m_qMutex);
     if (compareAndSet(m_record.refMetadata().refTrackInfo().ptrReplayGain(), replayGain)) {
         markDirtyAndUnlock(&lock);
-        emit replayGainUpdated(replayGain);
+        emit replayGainUpdated(replayGain, mixxx::ReplayGain::UpdateWhenStopped);
+    }
+}
+
+void Track::adjustReplayGainFromDeckGain(const QString& deckGroup) {
+    QMutexLocker lock(&m_qMutex);
+    mixxx::ReplayGain replayGain = m_record.getMetadata().getTrackInfo().getReplayGain();
+    auto deckPregainCO = ControlProxy(deckGroup, "pregain");
```

Comment:
For my feeling the Track Object should not access Controls. Can you just pass the gain adjustment and move the CO part to slotUpdateReplaygain(). 
I think you can also make use of ControlObject::get() a zero value indicated that there is nothing to do.

Reply:
I had that before, but basetrackplayer doesn't know which deck to use as the pregain reference. Therefore I will need a new signal that indicates which deck to use as the gain source instead of a replaygain value

Reply:
done